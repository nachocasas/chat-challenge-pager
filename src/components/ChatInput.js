import React, { Component } from 'react';
export default class ChatInput extends Component {

  constructor(props){
    super(props);
    this.state = {
      text: '',
    }
    this.tipingTimeOut = null;
    this.handleChange = this.handleChange.bind(this);
    this.handleEnter = this.handleEnter.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    let self = this;
    this.setState({
      text: e.target.value
    })

    this.props.isTyping(true)
    if(this.tipingTimeOut){ 
      clearTimeout(this.tipingTimeOut);
    }

    this.tipingTimeOut = setTimeout(function() { self.props.isTyping(false) }, 2000);
  }

  handleEnter(e){
    const code = (e.keyCode ? e.keyCode : e.which)
    if(code == 13){
      this.handleSubmit();
    }
  }

  handleSubmit(e){
    if(this.state.text.trim() != ''){
      this.props.onEnter(this.state.text);
      this.setState({ text: '' });
    }
  }

  render(){
    return(
      <div className="chatinput">
        <input onChange={ this.handleChange } onKeyPress={ this.handleEnter } value={this.state.text} />
        <button onClick={ this.handleSubmit }>Send</button>
      </div>
    )
  }
}