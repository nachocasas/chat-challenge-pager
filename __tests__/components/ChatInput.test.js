import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChatInput from '../../src/components/ChatInput';

describe('<ChatInput />', () => {
  describe('render()', () => {

    test('render component', () => {
      const wrapper = shallow(<ChatInput />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });
});