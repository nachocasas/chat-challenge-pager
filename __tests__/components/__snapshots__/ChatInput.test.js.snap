// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatInput /> render() render component 1`] = `
<div
  className="chatinput"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Send
  </button>
</div>
`;
